### PR TITLE
Properly build path to the mtl file when loading obj

### PR DESCRIPTION
**Pull request by Ang3lus**
Before patch:
```
MyIOStream(assets/raw/models/test_model/test_model.obj)
MyIOStream(test_model.mtl)
```

After patch:
```
MyIOStream(assets/raw/models/test_model/test_model.obj)
MyIOStream(assets/raw/models/test_model/test_model.mtl)
```

[Review on Reviewable](https://reviewable.io/reviews/assimp/assimp/672)

### CODE REVIEW

File: code/ObjFileImporter.cpp [hunk 1]
```diff
@@ -138,8 +138,8 @@ void ObjFileImporter::InternReadFile( const std::string& pFile, aiScene* pScene,
     if ( pos != std::string::npos ) {
         modelName = pFile.substr(pos+1, pFile.size() - pos - 1);
         folderName = pFile.substr( 0, pos );
-        if ( folderName.empty() ) {
-            pIOHandler->PushDirectory( folderName );
+        if ( !folderName.empty() ) {
+            pIOHandler->PushDirectory( folderName + "/");
```
Comment (Ang3lus) [inline code/ObjFileImporter.cpp]: Looked through the code and maybe this `+ "/"` should be placed in the ObjFileParser.cpp as there is the place where path is added:
```
...
    if ( m_pIO->StackSize() > 0 ) {
        const std::string &path = m_pIO->CurrentDirectory();
        absName = path + strMatName;
    } else {
        absName = strMatName;
    }
    IOStream *pFile = m_pIO->Open( absName );
...
```

What do You think ?

Comment (acgessler) [inline code/ObjFileImporter.cpp]: @kimkulling your call!

**Comment by Ang3lus** [comment]
* Used rfind and getOsSeparator to separate file from path
* Missing mtl path separator is added in the ObjFileParser.cpp when needed

**Comment by Ang3lus** [comment]
* Removed hardcoded path separator from one more place

**Comment by acgessler** [comment]
Looks good, two comments above.

**Comment by Ang3lus** [comment]
* Reverted back to plain '/'
* Added additional check for empty path

**Comment by Ang3lus** [comment]
Merge ? :)

**Comment by acgessler** [comment]
I think "\\/" is what we want, right? Rest looks good, but please revert this check.

**Comment by Ang3lus** [comment]
* Check for empty path removed

As for "\/" - why someone would want to escape this char ?

**Comment by acgessler** [comment]
This line is changed wrt to the original. It used to be `std::string::size_type pos = pFile.find_last_of( "\\/" );` and I'd prefer not changing it since we might encounter unanticipated regressions.

**Comment by Ang3lus** [comment]
Done.

**Comment by acgessler** [comment]
Thanks!

**Pull request merged**